### ISSUE TITLE
[maven] maven-pmd-plugin reports violations twice

### ISSUE THREAD

**Issue by aaime** [pmd/pmd#4001]
**Affects PMD Version:** 

Tested with latest, PMD 6.45 and maven-pmd-plugin 3.17.0. JDK 8 or 11, does not seem to make a difference.

**Description:**

The Maven violation report includes all violations twice, e.g.:

```
[INFO] --- maven-pmd-plugin:3.17.0:check (default) @ gs-wps-core ---
[INFO] PMD version: 6.45.0
[INFO] PMD Failure: org.geoserver.wps.WPSTestSupport:8 Rule:wildcards Priority:3 No Wildcard Imports.
[INFO] PMD Failure: org.geoserver.wps.WPSTestSupport:8 Rule:wildcards Priority:3 No Wildcard Imports.
```

**Steps to reproduce:**

Add any violation in the code, run the pmd checks in Maven. 
We are running the maven plugin the following way:

```xml
          <plugin>
            <groupId>org.apache.maven.plugins</groupId>
            <artifactId>maven-pmd-plugin</artifactId>
            <version>3.17.0</version>
            <executions>
              <execution>
                <goals>
                  <goal>check</goal>
                </goals>
              </execution>
            </executions>
            <dependencies>
              <dependency>
                <groupId>net.sourceforge.pmd</groupId>
                <artifactId>pmd-core</artifactId>
                <version>${pmd.version}</version>
              </dependency>
              <dependency>
                <groupId>net.sourceforge.pmd</groupId>
                <artifactId>pmd-java</artifactId>
                <version>${pmd.version}</version>
              </dependency>
              <dependency>
                <groupId>net.sourceforge.pmd</groupId>
                <artifactId>pmd-javascript</artifactId>
                <version>${pmd.version}</version>
              </dependency>
              <dependency>
                <groupId>net.sourceforge.pmd</groupId>
                <artifactId>pmd-jsp</artifactId>
                <version>${pmd.version}</version>
              </dependency>
            </dependencies>
            <configuration>
              <rulesets>
                <ruleset>${geoserverBaseDir}/../build/qa/pmd-ruleset.xml</ruleset>
                <ruleset>${geoserverBaseDir}/../build/qa/pmd-junit-ruleset.xml</ruleset>
              </rulesets>
              <failurePriority>3</failurePriority>
              <minimumPriority>3</minimumPriority>
              <verbose>true</verbose>
              <printFailingErrors>true</printFailingErrors>
              <includeTests>true</includeTests>
              <sourceEncoding>UTF-8</sourceEncoding>
              <outputEncoding>UTF-8</outputEncoding>
              <excludeRoots>
                <excludeRoot>target/generated-sources</excludeRoot>
              </excludeRoots>
            </configuration>
          </plugin>
```

Reference in the [full pom file here](https://github.com/geoserver/geoserver/blob/main/src/pom.xml#L2297).

**Comment by adangel** [comment]
Thanks for the report. I could reproduce and verify it with your project.
On further tests, I've seen that it is not related to the update of the versions - this even happens with m-pmd-p 3.14.0 and PMD 6.42.0

It turns out, that this is just a display issue - so PMD doesn't find duplicated violations and in the generated report pmd.xml, there are no duplicated violations. The duplication reporting only happens during logging on the console.

This is a side effect of using both options `verbose` and `printFailingErrors` (https://github.com/geoserver/geoserver/blob/810f995824452c7f4d6470dddd116e2d0e2f6ae9/src/pom.xml#L2373-L2374). These two options do almost the same:
* verbose prints warnings and failures
* printFailingErrors only prints failures
Failures in the sense, that the violations fail the build, because the priority of the violations exceed the failurePriority.

To fix the issue in your build, just remove the line `<printFailingErrors>true</printFailingErrors>`.

I've created an issue upstream (https://issues.apache.org/jira/browse/MPMD-344) as the issue is in the maven-pmd-plugin.
